src/main/java/tc/oc/pgm/match/MatchManagerImpl.java

```diff
@@ -158,7 +164,7 @@ public void unloadMatch(@Nullable String id) {
   @Override
   public Optional<Match> cycleMatch(@Nullable Match oldMatch, PGMMap nextMap, boolean retry) {
     // Pop map out
-    MapCommands.popNextMap();
+    pgmMapOrder.popNextMap();
```

Comment:
this is popping the map off twice, as the nextMap parameter passed here, should be the popped map.